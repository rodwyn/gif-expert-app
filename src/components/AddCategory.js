import React, { useState } from 'react'

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = event => {
    setInputValue(event.target.value);
  };
  const handleSubmit = event => { event.preventDefault(); };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </ form>
  )
}
