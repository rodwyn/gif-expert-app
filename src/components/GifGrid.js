import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'

import { GifGridItem } from './GifGridItem'

export const GifGrid = ({category}) => {
  const {data,loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__bounceInLeft">{category}</h3>
      { loading && <p>Loading</p> }
      <div className="card-grid">  
        {
          data.map(
            img => 
              <GifGridItem key={img.id} {...img} />
          )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
