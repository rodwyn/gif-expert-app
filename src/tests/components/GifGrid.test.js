import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('Testing the <GifGrid /> component.', () => {
  
  test('Should render the component correctly.', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true 
    });

    const wrapper = shallow(<GifGrid category='gorillaz' />);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('Should render items when useFetchGifs get images.', () => {
    const gifs = [{
      id: '123-a-456',
      url: 'rodwyn.com/sad.gif',
      title: 'a sad gif'
    },{
      id: '123-a-457',
      url: 'rodwyn.com/happy.gif',
      title: 'a happy gif'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category='gorillaz' />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
})
