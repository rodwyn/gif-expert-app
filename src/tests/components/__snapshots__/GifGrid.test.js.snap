// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <GifGrid /> component. Should render items when useFetchGifs get images. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    gorillaz
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123-a-456"
      key="123-a-456"
      title="a sad gif"
      url="rodwyn.com/sad.gif"
    />
    <GifGridItem
      id="123-a-457"
      key="123-a-457"
      title="a happy gif"
      url="rodwyn.com/happy.gif"
    />
  </div>
</Fragment>
`;

exports[`Testing the <GifGrid /> component. Should render the component correctly. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    gorillaz
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
